runtime/auth/http-auth-aws/common/src/aws/smithy/kotlin/runtime/http/auth/AwsHttpSigner.kt

```diff
@@ -135,6 +136,7 @@ public class AwsHttpSigner(private val config: Config) : HttpSigner {
             algorithm = config.algorithm
 
             region = when {
+                algorithm == AwsSigningAlgorithm.SIGV4_ASYMMETRIC && !configSigningRegionSet.isNullOrEmpty() -> configSigningRegionSet.joinToString(",")
```

Comment:
**suggestion**: Add the context here in a comment about _why_ config takes precedence. 

```kotlin
when {
     // siging region set from client config overrides all other sources like endpoints because it's designed as an
     // escape hatch for customers to control/limit which regions a request will be valid for (e.g. since some services
     // use '*')
     algorithm == AwsSigningAlgorithm.SIGV4_ASYMMETRIC ...
     ...
}
```